gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -137,13 +139,17 @@ Collection<CopyEntity> generateCopyEntities(FileSystem targetFs, CopyConfigurati
       // TODO: determine whether unnecessarily expensive to repeatedly re-create what should be the same FS: could it
       // instead be created once and reused thereafter?
       FileSystem actualSourceFs = getSourceFileSystemFromFileStatus(srcFileStatus, defaultHadoopConfiguration);
+      Path toPath = PathUtils.getRootPathParent(srcPath);
 
-      // TODO: Add preservation of ancestor ownership and permissions!
-
+      // preserving ancestor permissions till root between src and dest
```

Comment:
since `toPath` is one level beneath root, is that where perms are preserved to?  (that actually seems fine, so not suggesting to change anything other than inaccurate comment)